internal/healthchecks/builtin.go

```diff
@@ -0,0 +1,200 @@
+				gvkErrors = appendResourceError(gvkErrors, obj, "DaemonSet is not ready")
+			}
+		case *appsv1.ReplicaSet:
+			if err := client.Get(ctx, objectKey, obj); err != nil {
+				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
+				continue
+			}
+			if obj.Status.AvailableReplicas != obj.Status.Replicas {
+				gvkErrors = appendResourceError(gvkErrors, obj, "ReplicaSet is not ready")
+			}
+		case *corev1.Pod:
+			if err := client.Get(ctx, objectKey, obj); err != nil {
+				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
+				continue
+			}
+			if obj.Status.Phase != corev1.PodRunning && obj.Status.Phase != corev1.PodSucceeded {
+				gvkErrors = appendResourceError(gvkErrors, obj, "Pod is not Running or Succeeded")
+			}
+		case *apiregistrationv1.APIService:
+			if err := client.Get(ctx, objectKey, obj); err != nil {
+				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
+				continue
+			}
+
+			conditionExists := false
+			for _, condition := range obj.Status.Conditions {
+				if condition.Type == apiregistrationv1.Available {
+					if condition.Status != "True" {
+						gvkErrors = appendResourceError(gvkErrors, obj, condition.Message)
+					}
+					conditionExists = true
+					break
+				}
+			}
+			if conditionExists {
+				continue
+			}
+			// If we are here we didn't find the "Available" condition, so we assume the APIService is non healthy.
+			gvkErrors = appendResourceError(gvkErrors, obj, "Available condition not found")
+		case *apiextensionsv1.CustomResourceDefinition:
+			if err := client.Get(ctx, objectKey, obj); err != nil {
+				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
+				continue
+			}
+
+			conditionExists := false
+			for _, condition := range obj.Status.Conditions {
+				if condition.Type == apiextensionsv1.Established {
+					if condition.Status != "True" {
+						gvkErrors = appendResourceError(gvkErrors, obj, condition.Message)
+					}
+					conditionExists = true
+					break
+				}
+			}
+			if conditionExists {
+				continue
+			}
+			gvkErrors = appendResourceError(gvkErrors, obj, "Established condition not found")
+		default:
+			// If we don't know how to check the health of the object, we assume it's healthy.
+			continue
+		}
+	}
+
+	return errors.Join(gvkErrors...)
+}
+
+// toErrKey returns a string that identifies a resource based on its GVK and namespace/name. This key is used
+// to identify the resource in the error message.
+func toErrKey(resource client.Object) string {
+	group := resource.GetObjectKind().GroupVersionKind().Group
+	namespace := resource.GetNamespace()
+	if group == "" {
+		group = "core"
+	}
+
+	// If the resource is namespaced, include the namespace in the key.
+	if namespace != "" {
+		return fmt.Sprintf("(%s.%s.%s)(%s/%s)", resource.GetObjectKind().GroupVersionKind().Kind, group, resource.GetObjectKind().GroupVersionKind().Version, namespace, resource.GetName())
```

Comment:
Is there a particular reason we are building our own string format for conveying the GVK? It looks like the [`schema.GroupVersionKind` type returned by `resource.GetObjectKind().GroupVersionKind()` has a `String()` function that outputs the GVK in the format `{group}/{version}, Kind={kind}`.](https://github.com/kubernetes/apimachinery/blob/bc548d1d2406dffb44e1f6ea7ec1f4e9744cf170/pkg/runtime/schema/group_version.go#L161-L163)  I wonder if it might be best to reuse that format since that is in the apimachinery package and is a format that is used by Kubernetes (and I'm certain I've seen in other places). This seems like it might be a format users familiar with Kubernetes might be more used to and inherently easier to parse than an arbitrary format specified only by rukpak.

Reply:
I applied your proposed changes, now the format is: "(apps/v1, Kind=DaemonSet)(MyDaemonSet): DaemonSet is not ready"

Reply:
done